cmd/thanos/query_frontend.go

```diff
@@ -136,6 +136,8 @@ func registerQueryFrontend(app *extkingpin.App) {
 
 	cmd.Flag("query-frontend.forward-header", "List of headers forwarded by the query-frontend to downstream queriers, default is empty").PlaceHolder("<http-header-name>").StringsVar(&cfg.ForwardHeaders)
 
+	cmd.Flag("query-frontend.num-shards", "Number of queriers to use when sharding PromQL queries").IntVar(&cfg.NumShards)
```

Comment:
We might use a more descriptive flag here

Reply:
Is it possible to specify the number of shards to use for each request? `query-frontend.num-shards` this flag can be a default fallback value if no shard num is specified from the query.

Reply:
Yeah, do we have any plan to dynamically specify the number of shards to use for each query? If not specified we can default to this flag.

Reply:
The number of shards will be predefined and static, we don't have the option of tweaking the sharding number dynamically.

Reply:
I modified the description a bit to make things a bit more clear